src/main/java/dook/services/Parser.java

```diff
@@ -0,0 +1,177 @@
+package dook.services;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import dook.DookException;
+import dook.command.*;
+import dook.task.Deadline;
+import dook.task.Event;
+import dook.task.Task;
+import dook.task.Todo;
+
+/**
+ * Responsible for parsing plaintext commands and returning appropriate
+ * Command objects.
+ */
+public class Parser {
+    private static CommandInfo parseKeyword(String keyword) {
+        try {
+            return CommandInfo.valueOf(keyword);
+        } catch (IllegalArgumentException e) {
+            return CommandInfo.invalid;
+        }
+    }
+
+    /**
+     * Takes a string input and parses it, generating the correct Command
+     * object type with the relevant data from user input.
+     *
+     * @param input The raw string input from the user.
+     * @return Command object parsed out from the input.
+     * @throws DookException Exception thrown by Dook.
+     */
+    public Command parseFullInput(String input) throws DookException {
+        String[] tmp = input.split(" ", 2);
+        CommandInfo command = parseKeyword(tmp[0].trim());
+        String body = tmp.length == 1 ? "" : tmp[1].trim();
+        switch (command) {
+        case bye:
+            return new ByeCommand();
+        case list:
+            return new ListCommand();
+        case save:
+            return new SaveCommand();
+        case mark:
+            return handleMark(body, true);
+        case unmark:
+            return handleMark(body, false);
+        case todo:
+            return handleToDo(body);
+        case deadline:
+            return handleDeadline(body);
+        case event:
+            return handleEvent(body);
+        case delete:
+            return handleDelete(body);
+        case after:
+            return handleAfter(body);
+        case before:
+            return handleBefore(body);
+        case during:
+            return handleDuring(body);
+        case find:
+            return handleFind(body);
+        default:
+            return new InvalidCommand();
+        }
+    }
+    private Command handleMark(String body, boolean value) throws DookException {
```

Comment:
Good job on following the convention of starting these methods which do things as a verb.